draft-ietf-rtcweb-security-arch.xml

```diff
@@ -556,6 +556,163 @@
+             or later during the session.
+	       </t>
+           <section title="Offer Procedures" anchor="sec.sdp-id-attr-oa-add-off">
+             <t>
+               When an offerer sends an offer, in order to provide its identity assertion to the peer, it MUST include an 
+               'identity' attribute in the offer. In addition, it MUST include SDP 'fingerprint' attributes
+               for each fingerprint that have been used to assert the identity.
+             </t>
+             <t>
+               When the offerer receives an answer that contains an 'identity' attribute, the offerer can use
+               the attribute information to contact the IdP, and verify the identity of, the peer. If the offerer does not
+               accept the attribute (or the attribute value), or if the identity verification fails, it MUST 
+               take proper actions to reject the answer.
+             </t>
+           </section>
+           <section title="Answerer Procedures" anchor="sec.sdp-id-attr-oa-add-ans">
+             <t>
+               When an answerer receives in offer that contains an 'identity' attribute, the answerer can use the
+               the attribute information to contact the IdP, and verify the identity of, the peer. If the answerer 
+               does not accept the attribute value, or if the identity verification fails, it MUST reject the offer.
+             </t>
+             <t>
+               When an answerer sends an answer, in order to provide its identity assertion to the peer, it MUST include an 
+               'identity' attribute in the answer. In addition, it MUST include 'fingerprint' attributes
+               for each fingerprint that have been used to assert the identity.
+             </t>
+             <t>
+               The answerer can include an 'identity' attribute in the answer independently of whether the 
+               corresponding offer contained an identity attribute or not.
+             </t>
+            </section>
+          </section>
+	      <section title="Modifying Identity Assertion within Session Description" anchor="sec.sdp-id-attr-oa-mod">
+	        <t>
+              When an endpoint has included its identity assertion in a previous offer or answer, and when the endpoint later in session
+              sends an offer or answer, the endpoint can either keep the identity assertion value unchanged, modify the value
+              or revoke the identity assertion. This section describes the procedures for that.
+            </t>
+            <t>
+              NOTE: Specific usages might restrict the modification and/or revoking of the identity assertion within a session.
+	        </t>
+            <section title="Offer Procedures" anchor="sec.sdp-id-attr-oa-mod-off">
+              <t>
+                When the offerer sends an offer, if the offerer wants to maintain the identity assertion value it previously provided (in an offer
```

Comment:
This section (and subsections) is a little wordy.  It being so repetitive is a potential source of problems.  How about consolidating the two offer and answer sections into:

> When an offerer or answerer creates a new offer or answer that modifies a successfully negotiated session, they can retain the same 'identity' attribute value, change or remove the value.  The rules for processing an offer or answer that modifies a session are functionally the same as during session creation, though there could be policies that cause offers or answers to be rejected if the validated identity changes.
>
> Adding, removing or changing the 'identity' attribute in a new offer or answer could change the value of the identity that a peer associates with the session.  Some usages might proscribe the sorts of change that can be supported when the identity of a peer changes.  For instance, `<xref target="???">`WebRTC`</xref>` requires that once an identity is successfully validated, all subsequent offers or answers are rejected unless the identity assertion can be validated successfully and the identity of the peer does not change.

Reply:
I am ok to add the offerer and answerer procedures in a single section. However, in your suggestion above you again assume that something happened during session creating (read: initial offer), which is what we are trying to get away from. What is important is that the attribute has previously been added in an offer/answer exchange within the session. Whether that exchange was the initial of the session or not is irrelevant.

Of course, certain usages/architectures might mandate the identity attribute to be added in the initial offer/answer of the session, but in that case those architectures must document such restrictions.

Reply:
Other documents have taken to clarifying that "initial offer" means "the first offer that this attribute is included in" rather than "the first offer in the session." I beleive you'll find that doing so here makes your life much easier.

Reply:
At the previous MMUSIC meeting we discussed a new structure for O/A procedures, to get rid of the "initial" confusion, and the text is based on that discussion.

Having said that, if we want to use the "old way", with a clarification, I can do that.